lib/user.rb

```diff
@@ -0,0 +1,27 @@
+class User
+  attr_reader :name
+  attr_accessor :jokes, :learned
+  def initialize (name)
+    @name = name
+    @jokes = []
+    @learned = false
+  end
+
+  def jokes(joke)
+    if learned == true
```

Comment:
It looks like you got a little ahead of yourself with the `@learned` variable. This is why it's important to do one test and one interaction pattern at a time. There was no need for the `@learned` variable a this point, but we wanted the `learn` method to do the shoveling of the joke argument into the `@jokes` array.